lib/types/src/schema/node.ts

```diff
@@ -2,7 +2,9 @@ import { SchemaOutput } from '../schema'
 
 export const NodeSchema = {
   entry: 'string?',
-  args: 'array.string?'
+  args: 'array.string?',
+  useVault: 'boolean?',
```

Comment:
actually in an ideal world wouldn't it be cool(?) if Vault were a separate plugin that got env vars and put them in state? then you could have other plugins that did that for variables from other sources. would be a breaking change, apps (or the preset i guess) would have to do

```yml
hooks:
  run:local:
    - VaultEnv
    - Node
```

but i think that coudl be really powerful

Reply:
i guess that would break the convention of only keeping vault vars in memory and not writing them to a (`.env`/state) file

Reply:
I think `useVault` is less confusing without the negative so let's keep that; `useVault: false` reads fine to me.

Yeah it would be good to uncouple Vault from these plugins and write all the state into a file (though that might be a security issue...? I guess `.env` files aren't usually considered that bad though?) but it probably makes sense to wait until we come across an alternative before doing the work. If cyber security move forward with https://www.doppler.com then that could justify a big major release to support it, though for now Vault is the source of truth for configuration values in Customer Products imo